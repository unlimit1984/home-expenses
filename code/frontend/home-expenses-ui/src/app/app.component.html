<ng-container *ngIf="layoutVersion === LayoutVersion.AngularMaterial">
  <mat-toolbar color="primary" class="desktop-toolbar">
    <a class="navbar-brand" routerLinkActive="active" routerLink="/">Home Expenses</a>
    <div
      *ngIf="tokenAuthService.isValidAccessToken()"
      class="d-flex flex-grow-1 align-items-center justify-content-end"
    >
      <a *ngIf="!isMobile" routerLinkActive="active" [routerLink]="['/settings']" class="desktop-settings-link me-2">
        <div class="d-flex flex-column align-items-center">
          <mat-icon matTooltip="Settings" [matTooltipPosition]="'below'">settings</mat-icon>
        </div>
      </a>
      <button type="button" class="btn btn-danger" (click)="signout()">Sign out</button>
    </div>
  </mat-toolbar>

  <div class="page-wrapper">
    <mat-sidenav-container
      *ngLet="tokenAuthService.isValidAccessToken(); let isValidAccessToken"
      class="desktop-sidenav-container"
    >
      <mat-sidenav
        class="desktop-sidenav"
        [style.width.px]="isMobile ? 0 : isSidenavCollapsed ? collapsedWidth : expandedWidth"
        mode="side"
        [opened]="true"
        [autoFocus]="false"
      >
        <mat-nav-list class="nav-list">
          <a
            mat-list-item
            routerLinkActive="active-link"
            [routerLink]="isValidAccessToken ? ['/all-expenses'] : null"
            [disabled]="!isValidAccessToken"
          >
            <ng-container [ngSwitch]="isSidenavCollapsed">
              <ng-container *ngSwitchCase="false">
                <div class="menu-item-content-expanded-desktop">
                  <mat-icon>payments</mat-icon>
                  <span class="ps-2">All expenses</span>
                </div>
              </ng-container>

              <ng-container *ngSwitchCase="true">
                <div class="menu-item-content-expanded-desktop">
                  <mat-icon matTooltip="All expenses" [matTooltipPosition]="'right'">payments</mat-icon>
                </div>
              </ng-container>
            </ng-container>
          </a>

          <a
            mat-list-item
            routerLinkActive="active-link"
            [routerLink]="isValidAccessToken ? ['/expenses'] : null"
            [disabled]="!isValidAccessToken"
          >
            <ng-container [ngSwitch]="isSidenavCollapsed">
              <ng-container *ngSwitchCase="false">
                <div class="menu-item-content-expanded-desktop">
                  <mat-icon>paid</mat-icon>
                  <span class="ps-2">Expenses</span>
                </div>
              </ng-container>

              <ng-container *ngSwitchCase="true">
                <div class="menu-item-content-expanded-desktop">
                  <mat-icon matTooltip="Expenses (Simple)" [matTooltipPosition]="'right'">paid</mat-icon>
                </div>
              </ng-container>
            </ng-container>
          </a>

          <div class="last-menu-div">
            <a mat-list-item (click)="toggleSidenav()">
              <ng-container [ngSwitch]="isSidenavCollapsed">
                <ng-container *ngSwitchCase="false">
                  <div class="menu-item-content-expanded-desktop">
                    <mat-icon>keyboard_double_arrow_left</mat-icon>
                    <span class="ps-2">Collapse</span>
                  </div>
                </ng-container>

                <ng-container *ngSwitchCase="true">
                  <mat-icon matTooltip="Expand" [matTooltipPosition]="'right'">keyboard_double_arrow_right</mat-icon>
                </ng-container>
              </ng-container>
            </a>
          </div>
        </mat-nav-list>
      </mat-sidenav>

      <mat-sidenav-content [style.margin-left.px]="isMobile ? 0 : isSidenavCollapsed ? collapsedWidth : expandedWidth">
        <router-outlet></router-outlet>
      </mat-sidenav-content>
    </mat-sidenav-container>
  </div>

  <mat-toolbar class="mobile-toolbar">
    <mat-nav-list class="mobile-nav-list">
      <a
        class="p-0 flex-grow-1"
        mat-list-item
        routerLinkActive="active-link"
        [routerLink]="tokenAuthService.isValidAccessToken() ? ['/all-expenses'] : null"
        [disabled]="!tokenAuthService.isValidAccessToken()"
      >
        <div class="d-flex flex-column align-items-center">
          <mat-icon>payments</mat-icon>
          <span>All expenses</span>
        </div>
      </a>
      <a
        class="p-0 flex-grow-1"
        mat-list-item
        routerLinkActive="active-link"
        [routerLink]="tokenAuthService.isValidAccessToken() ? ['/expenses'] : null"
        [disabled]="!tokenAuthService.isValidAccessToken()"
      >
        <div class="d-flex flex-column align-items-center">
          <mat-icon>paid</mat-icon>
          <span>Expenses (Simple)</span>
        </div>
      </a>
      <a
        class="p-0 flex-grow-1"
        mat-list-item
        routerLinkActive="active-link"
        [routerLink]="tokenAuthService.isValidAccessToken() ? ['/settings'] : null"
        [disabled]="!tokenAuthService.isValidAccessToken()"
      >
        <div class="d-flex flex-column align-items-center">
          <mat-icon>settings</mat-icon>
          <span>Settings</span>
        </div>
      </a>
    </mat-nav-list>
  </mat-toolbar>
</ng-container>

<ng-container *ngIf="layoutVersion === LayoutVersion.Standard">
  <ng-container *ngIf="!configService.config.featureFlags.multiTabMode; else multiTabSolution">
    <app-header></app-header>
    <router-outlet></router-outlet>
  </ng-container>

  <ng-template #multiTabSolution>
    <ng-container [ngSwitch]="isAppAvailable">
      <ng-container *ngSwitchCase="true">
        <app-header></app-header>
        <router-outlet></router-outlet>
      </ng-container>

      <div *ngSwitchCase="false">
        <p data-test-id="warning">WARNING: Multi tab usage is not allowed.</p>
        <p>You can see that warning because you are trying to open App in another tab.</p>
        <p>If you are sure there is no any opened tab you will be reloaded in {{ timeLeft }}s...</p>
      </div>
    </ng-container>
  </ng-template>
</ng-container>
