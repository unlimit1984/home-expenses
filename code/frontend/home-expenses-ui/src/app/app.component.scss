.active-link {
  background-color: #ccc; /* Change to desired highlight color */
}

.sidenav-container {
  display: flex;
  flex-direction: column;
  height: calc(100vh - 64px);
}

.sidenav-content {
  display: flex;
  flex: 1; /* Fill remaining space */
  flex-direction: column;
}

.main-content {
  flex: 1; /* Fill remaining space */
  overflow-y: auto; /* Enable scrolling if content exceeds available height */
}
