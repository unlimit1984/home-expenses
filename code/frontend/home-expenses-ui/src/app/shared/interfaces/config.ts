export interface IConfig {
  readonly APP_VERSION: string;
  readonly API: string;
}
