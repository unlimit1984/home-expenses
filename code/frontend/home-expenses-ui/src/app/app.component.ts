/*
 * Author: Vladimir Vysokomornyi
 */

import { Component, inject, OnDestroy, OnInit } from '@angular/core';
import { ConfigService } from './services/config/config.service';
import { LayoutVersion } from './shared/interfaces/layout.version';
import { TokenAuthService } from './services/token-vault/token-auth.service';
import { Store } from '@ngrx/store';
import { signout } from './store/auth/auth.actions';
import { BreakpointObserver } from '@angular/cdk/layout';
import { TranslateService } from '@ngx-translate/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent implements OnInit, OnDestroy {
  protected readonly layoutVersion: LayoutVersion;
  protected readonly LayoutVersion = LayoutVersion;
  isSidenavCollapsed = true;
  collapsedWidth = 64; // Adjust as needed
  expandedWidth = 200; // Adjust as needed
  isMobile: boolean = false;

  public configService = inject(ConfigService);

  private syncTabsInterval: string | number | NodeJS.Timeout;
  private autoReloadTimer: string | number | NodeJS.Timeout;
  public timeLeft = 3;
  public isAppAvailable = false;

  public tokenAuthService = inject(TokenAuthService);
  public translateService = inject(TranslateService);

  constructor(private store: Store, private breakpointObserver: BreakpointObserver) {
    this.layoutVersion = this.configService.config.featureFlags.layoutVersion;

    if (this.configService.config.featureFlags.multiTabMode) {
      this.removeOldAgeTab();
      this.makeAppCurrentOrBlockWithReloadTimer();
    }

    this.translateService.use(this.configService.getSelectedLanguage());
  }

  ngOnInit(): void {
    this.breakpointObserver.observe(['(max-width: 768px)']).subscribe((result) => {
      this.isMobile = result.matches;
    });
  }

  signout() {
    this.store.dispatch(signout());
  }

  private removeOldAgeTab() {
    const dateAsString: string = localStorage.getItem('tab');
    if (dateAsString && dateAsString.length > 0) {
      if (Number(dateAsString) < Date.now() - 2000) {
        console.log('Old tab found, will be deleted from localStorage');
        localStorage.removeItem('tab');
      } else {
        localStorage.setItem('tab', Date.now().toString());
      }
    }
  }

  private makeAppCurrentOrBlockWithReloadTimer() {
    if (this.syncTabsInterval) {
      clearInterval(this.syncTabsInterval);
    }
    if (this.autoReloadTimer) {
      clearInterval(this.autoReloadTimer);
    }

    const tab: string = localStorage.getItem('tab');
    if (!tab) {
      this.isAppAvailable = true;
      localStorage.setItem('tab', Date.now().toString());
      this.syncTabsInterval = setInterval(() => {
        localStorage.setItem('tab', Date.now().toString());
      }, 1000);
    } else {
      this.isAppAvailable = false;
      this.autoReloadTimer = setInterval(async () => {
        if (this.timeLeft > 0) {
          this.timeLeft--;
        } else {
          window.location.reload();
        }
      }, 1000);
    }
  }

  ngOnDestroy(): void {
    localStorage.removeItem('tab');
    clearInterval(this.syncTabsInterval);
    clearInterval(this.autoReloadTimer);
  }

  toggleSidenav() {
    this.isSidenavCollapsed = !this.isSidenavCollapsed;
  }
}
