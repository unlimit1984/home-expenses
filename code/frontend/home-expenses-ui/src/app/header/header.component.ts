/*
 * Author: Vladimir Vysokomornyi
 */

import { Component, inject } from '@angular/core';
import { Store } from '@ngrx/store';
import { signout } from '../store/auth/auth.actions';
import { ConfigService } from '../services/config/config.service';
import { TokenAuthService } from '../services/token-vault/token-auth.service';

@Component({
  selector: 'app-header',
  templateUrl: './header.component.html',
  styleUrls: ['./header.component.scss']
})
export class HeaderComponent {
  public tokenAuthService = inject(TokenAuthService);

  constructor(private store: Store, public configService: ConfigService) {}

  signout() {
    this.store.dispatch(signout());
  }
}
