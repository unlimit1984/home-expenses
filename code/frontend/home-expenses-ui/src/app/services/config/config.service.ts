/*
 * Author: Vladimir Vysokomornyi
 */

import { inject, Injectable } from '@angular/core';
import { IConfig } from '../../shared/interfaces/config';
import { catchError, Observable, tap } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { Store } from '@ngrx/store';
import { refreshTokensTick } from '../../store/auth/auth.actions';
import { TranslateService } from '@ngx-translate/core';
import { Language } from '../../shared/enums/language';

@Injectable({
  providedIn: 'root'
})
export class ConfigService {
  private translateService = inject(TranslateService);
  private _config: IConfig = null;
  get config() {
    return { ...this._config };
  }
  constructor(private http: HttpClient, private store: Store) {
    console.log('==constructor ConfigService');
  }

  private _initialize(config: IConfig) {
    if (!this._config) {
      console.log('==initialize ConfigService');
      this._config = config;
    } else {
      console.warn('Config was already initialized earlier. Should be initialized just once');
    }
  }

  public loadConfig(): Observable<IConfig> {
    console.log('==loadConfig');
    return this.http.get('./assets/config/config.json').pipe(
      tap((response: IConfig) => {
        this._initialize(response);
        this.store.dispatch(refreshTokensTick());
      }),
      catchError((err) => {
        console.warn(`Can't load config`);
        throw err;
      })
    );
  }

  public getSelectedLanguage(): string {
    const lang: string = localStorage.getItem('selectedLanguage');
    if (lang && Object.keys(Language).includes(lang)) {
      return lang;
    }

    localStorage.setItem('selectedLanguage', Language.en);
    return Language.en;
  }

  public setSelectedLanguage(lang: string): void {
    if (lang && Object.keys(Language).includes(lang)) {
      this.translateService.use(lang);
      localStorage.setItem('selectedLanguage', Language[lang]);
    }
  }
}
