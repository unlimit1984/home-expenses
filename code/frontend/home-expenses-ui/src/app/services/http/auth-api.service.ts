/*
 * Author: Vladimir Vysokomornyi
 */

import { inject, Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { User } from '../../shared/interfaces/user';
import { Credentials } from '../../shared/interfaces/credentials';
import { Tokens } from '../../interfaces/tokens';
import { ConfigService } from '../config/config.service';

@Injectable({
  providedIn: 'root'
})
export class AuthApiService {
  private configService: ConfigService = inject(ConfigService);

  constructor(private httpClient: HttpClient) {
    console.log('==constructor AuthApiService, api', this.configService.config.API);
  }

  public signupStart(credentials: Credentials): Observable<User> {
    return this.httpClient.post<User>(`${this.configService.config.API}/auth/signup-start`, credentials);
  }
  public signupFinish(credentials: Credentials): Observable<User> {
    return this.httpClient.post<User>(`${this.configService.config.API}/auth/signup-finish`, credentials);
  }

  public signin(credentials: Credentials): Observable<Tokens> {
    return this.httpClient.post<Tokens>(`${this.configService.config.API}/auth/signin`, credentials);
  }

  public signout(): Observable<boolean> {
    // return timer(2000).pipe(map(() => true));
    return this.httpClient.get<boolean>(`${this.configService.config.API}/auth/signout`);
  }

  public refreshTokens(): Observable<Tokens> {
    return this.httpClient.post<Tokens>(`${this.configService.config.API}/auth/refresh-token`, null);
  }
}
