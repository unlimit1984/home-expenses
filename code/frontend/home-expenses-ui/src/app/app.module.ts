/*
 * Author: Vladimir Vysokomornyi
 */

import { APP_INITIALIZER, NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { ExpensesEffects } from './store/expenses/expenses-effects';
import { HTTP_INTERCEPTORS, HttpClient, HttpClientModule } from '@angular/common/http';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { ReactiveFormsModule } from '@angular/forms';
import { reducers } from './store/reducers';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { environment } from '../environments/environment';
import { ExpensesComponent } from './pages/expenses/expenses.component';
import { AllExpensesComponent } from './pages/all-expenses/all-expenses.component';
import { SignupComponent } from './pages/signup/signup.component';
import { SigninComponent } from './pages/signin/signin.component';
import { HeaderComponent } from './header/header.component';
import { SignupFinishComponent } from './pages/signup-finish/signup-finish.component';
import { ErrorPageComponent } from './pages/error-page/error-page.component';
import { AuthEffects } from './store/auth/auth.effects';
import { JwtInterceptorService } from './services/interceptor/jwt-interceptor.service';
import { ConfigService } from './services/config/config.service';
import { Observable } from 'rxjs';
import { IConfig } from './shared/interfaces/config';
import { TestComponent } from './test/test.component';
import { StoreRouterConnectingModule } from '@ngrx/router-store';
import { RouterSerializer } from './store/router/router.reducer';
import { provideAnimationsAsync } from '@angular/platform-browser/animations/async';
import { MatSidenavModule } from '@angular/material/sidenav';
import { MatIconModule } from '@angular/material/icon';
import { MatListModule } from '@angular/material/list';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatMenuModule } from '@angular/material/menu';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { MatButtonModule } from '@angular/material/button';
import { MatCardModule } from '@angular/material/card';
import { MatTabsModule } from '@angular/material/tabs';
import { MatInputModule } from '@angular/material/input';
import { MatTooltipModule } from '@angular/material/tooltip';
import { MatTableModule } from '@angular/material/table';
import { MatPaginatorModule } from '@angular/material/paginator';
import { MatSortModule } from '@angular/material/sort';
import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
import { MatStepperModule } from '@angular/material/stepper';
import { MatDialogModule } from '@angular/material/dialog';
import { MatSelectModule } from '@angular/material/select';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { DragDropModule } from '@angular/cdk/drag-drop';
import { MatRadioModule } from '@angular/material/radio';
import { MatSliderModule } from '@angular/material/slider';
import { MatDatepickerModule } from '@angular/material/datepicker';
import { DateAdapter, MAT_DATE_LOCALE, MatNativeDateModule } from '@angular/material/core';
import { MatGridListModule } from '@angular/material/grid-list';
import { MatTreeModule } from '@angular/material/tree';
import { ScrollingModule } from '@angular/cdk/scrolling';
import { NgLetModule } from 'ng-let';
import { SettingsComponent } from './pages/settings/settings.component';
import { TranslateLoader, TranslateModule, TranslateService } from '@ngx-translate/core';
import { TranslateHttpLoader } from '@ngx-translate/http-loader';
import { BaseChartDirective, provideCharts, withDefaultRegisterables } from 'ng2-charts';

@NgModule({
  declarations: [
    AppComponent,
    ExpensesComponent,
    AllExpensesComponent,
    SettingsComponent,
    SignupComponent,
    SigninComponent,
    HeaderComponent,
    SignupFinishComponent,
    ErrorPageComponent,
    TestComponent
  ],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    HttpClientModule,
    AppRoutingModule,
    StoreModule.forRoot(reducers),
    EffectsModule.forRoot([ExpensesEffects, AuthEffects]),
    StoreDevtoolsModule.instrument({ logOnly: environment.prod, connectInZone: true }),
    StoreRouterConnectingModule.forRoot({
      serializer: RouterSerializer
    }),
    TranslateModule.forRoot({
      loader: {
        provide: TranslateLoader,
        useFactory: createTranslateLoader,
        deps: [HttpClient]
      }
    }),
    NgbModule,
    NgLetModule,
    ReactiveFormsModule,
    MatMenuModule,
    MatButtonModule,
    MatIconModule,
    MatCardModule,
    MatTabsModule,
    MatSidenavModule,
    MatListModule,
    MatToolbarModule,
    MatTooltipModule,
    MatInputModule,
    MatTableModule,
    MatPaginatorModule,
    MatSortModule,
    MatProgressSpinnerModule,
    MatStepperModule,
    MatDialogModule,
    MatSelectModule,
    MatCheckboxModule,
    DragDropModule,
    MatRadioModule,
    MatSliderModule,
    MatDatepickerModule,
    MatNativeDateModule,
    MatGridListModule,
    MatTreeModule,
    ScrollingModule,
    BaseChartDirective
  ],
  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: JwtInterceptorService,
      multi: true
    },
    {
      provide: APP_INITIALIZER,
      useFactory: initializeAppFactory,
      multi: true,
      deps: [ConfigService]
    },
    provideAnimationsAsync(),
    // { provide: MAT_DATE_LOCALE, useValue: 'en-GB' },
    // { provide: DateAdapter, useClass: MondayAsFirstDayOfWeekAdapter }
    provideCharts(withDefaultRegisterables())
  ],
  bootstrap: [AppComponent]
})
export class AppModule {
  constructor(private translateService: TranslateService, private configService: ConfigService) {
    this.translateService.use(this.configService.getSelectedLanguage());
  }
}

function initializeAppFactory(configService: ConfigService): () => Observable<IConfig> {
  return () => configService.loadConfig();
}

export function createTranslateLoader(http: HttpClient) {
  return new TranslateHttpLoader(http, './assets/i18n/', '.json');
}
