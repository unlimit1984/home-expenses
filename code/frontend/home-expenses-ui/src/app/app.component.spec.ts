/*
 * Author: Vladimir Vysokomornyi
 */

import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';
import { provideMockStore } from '@ngrx/store/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { ConfigService } from './services/config/config.service';
import { HeaderComponent } from './header/header.component';
import { By } from '@angular/platform-browser';

const mockConfigService: any = {
  config: {
    featureFlags: { multiTabMode: true }
  }
};
describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;
  let componentRender: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [AppComponent, HeaderComponent],
      imports: [RouterTestingModule, ReactiveFormsModule, HttpClientTestingModule],
      providers: [
        {
          provide: ConfigService,
          useValue: mockConfigService
        },
        provideMockStore({})
      ]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
    componentRender = fixture.nativeElement;
  });

  it('should create the app', () => {
    fixture.detectChanges();
    expect(component).toBeTruthy();
    expect(componentRender).toMatchSnapshot();
  });

  it('app should be available', () => {
    component.isAppAvailable = true;

    fixture.detectChanges();
    expect(componentRender).toMatchSnapshot();

    const warning = fixture.debugElement.query(By.css('[data-test-id="warning"]'));
    expect(warning).toBeNull();
  });

  xit('app should be unavailable', () => {
    component.isAppAvailable = false;

    fixture.detectChanges();
    expect(componentRender).toMatchSnapshot();

    const warning = fixture.debugElement.query(By.css('[data-test-id="warning"]'));
    expect(warning).toBeTruthy();
  });
});
