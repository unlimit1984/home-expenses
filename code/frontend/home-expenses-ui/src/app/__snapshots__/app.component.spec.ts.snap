// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent app should be available 1`] = `
<div
  id="root1"
>
  <app-header>
    <nav
      class="navbar navbar-expand-lg bg-light"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
          routerlink="/"
          routerlinkactive="active"
        >
          Home Expenses
        </a>
        <span
          class="version"
        >
          (ver. )
        </span>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarNavAltMarkup"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          
        </div>
      </div>
    </nav>
  </app-header>
  <router-outlet />
  
  
  
  
  
  
  
</div>
`;

exports[`AppComponent app should be unavailable 1`] = `
<div
  id="root2"
>
  
  <div>
    <p
      data-test-id="warning"
    >
      WARNING: Multi tab usage is not allowed.
    </p>
    <p>
      You can see that warning because you are trying to open App in another tab.
    </p>
    <p>
      If you are sure there is no any opened tab you will be reloaded in 3s...
    </p>
  </div>
  
  
  
  
</div>
`;

exports[`AppComponent should create the app 1`] = `
<div
  id="root0"
>
  <app-header>
    <nav
      class="navbar navbar-expand-lg bg-light"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
          routerlink="/"
          routerlinkactive="active"
        >
          Home Expenses
        </a>
        <span
          class="version"
        >
          (ver. )
        </span>
        <button
          aria-controls="navbarNavAltMarkup"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarNavAltMarkup"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse"
          id="navbarNavAltMarkup"
        >
          
        </div>
      </div>
    </nav>
  </app-header>
  <router-outlet />
  
  
  
  
  
  
  
</div>
`;
